### PR TITLE
crypto: fix several documentation issues

### PR DESCRIPTION
**Pull request by mtausig**
Some functions in the crypto module had a missing or unprecise return value documentation. I have added or fixed those (plus a few typos)

### CODE REVIEW

**Comment by mtausig** [comment]
Done.
I found a bug in `aes_init`while doing so (no error message for a bad key size).

**Comment by mtausig** [comment]
Ping

**Comment by PeterKietzmann** [comment]
@bergzand approve?

**Comment by bergzand** [comment]
@mtausig Could you please split the style changes into a separate PR and keep this PR focussed on documentation fixes? For me that really helps with the review process as both are easy to merge when they are separate PRs.

File: sys/include/crypto/aes.h [hunk 1]
```diff
@@ -81,8 +81,8 @@ typedef struct {
  * @param       key       a pointer to the key
  *
  * @return  CIPHER_INIT_SUCCESS if the initialization was successful.
- *          The command may be unsuccessful if the key size is not valid.
- *          CIPHER_ERR_BAD_CONTEXT_SIZE if CIPHER_MAX_CONTEXT_SIZE has not been defined (which means that the cipher has not been included in the build)
+ * @return  CIPHER_ERR_BAD_CONTEXT_SIZE if CIPHER_MAX_CONTEXT_SIZE has not been defined
```
Comment (bergzand) [inline sys/include/crypto/aes.h]: Could you please split this line into two to keep it below 80 chars width.

File: sys/include/crypto/ciphers.h [hunk 2]
```diff
@@ -125,8 +125,8 @@ typedef struct {
  * @param key_size   length of the encryption key
  *
  * @return  CIPHER_INIT_SUCCESS if the initialization was successful.
- *          The command may be unsuccessful if the key size is not valid.
- *          CIPHER_ERR_BAD_CONTEXT_SIZE if CIPHER_MAX_CONTEXT_SIZE has not been defined (which means that the cipher has not been included in the build)
+ * @return  CIPHER_ERR_BAD_CONTEXT_SIZE if CIPHER_MAX_CONTEXT_SIZE has not been defined (which means that the cipher has not been included in the build)
```
Comment (bergzand) [inline sys/include/crypto/ciphers.h]: And this one to keep it below 80 chars width

**Comment by mtausig** [comment]
Removed the code style fixes from this pull request.

File: sys/include/crypto/aes.h [hunk 3]
```diff
@@ -41,11 +41,11 @@ typedef uint8_t u8;
...
 # define GETU32(pt) (((u32)(pt)[0] << 24) ^ ((u32)(pt)[1] << 16) ^ \
-                             ((u32)(pt)[2] <<  8) ^ ((u32)(pt)[3]))
+                     ((u32)(pt)[2] <<  8) ^ ((u32)(pt)[3]))
 # define PUTU32(ct, st) { (ct)[0] = (u8)((st) >> 24); \
-                              (ct)[1] = (u8)((st) >> 16); \
-                              (ct)[2] = (u8)((st) >>  8); \
-                              (ct)[3] = (u8)(st); }
+                          (ct)[1] = (u8)((st) >> 16); \
+                          (ct)[2] = (u8)((st) >>  8); \
+                          (ct)[3] = (u8)(st); }
```
Comment (bergzand) [inline sys/include/crypto/aes.h]: This is going to cause a conflict with #10347

**Comment by bergzand** [comment]
ACK, please squash!

About the remaining style change mentioned above, If you can remove it from this PR, we can first merge #10347 and then this one immediately after.

**Comment by mtausig** [comment]
Yes. I'll rebase this patch after #10347 is merged.

File: sys/include/crypto/aes.h (hunk 3 above)
Comment (mtausig) [reply sys/include/crypto/aes.h]: (Un)done 7542e03

**Comment by bergzand** [comment]
@mtausig Could you please squash this into a single commit?

**Comment by miri64** [comment]
> @mtausig Could you please squash this into a single commit?

From having a quick look at the changes, I rather prefer to have several commits ordered by theme (fix line length, fix whitespace errors, fix spelling/grammar, etc.)

**Comment by miri64** [comment]
I'm alright with it, but since @bergzand did the actual review, I give him the final decision on that.

**Comment by bergzand** [comment]
Good enough for me, especially with the extended description of the second commit. Thanks for taking care of this!

**Pull request merged**